include/openthread/platform/toolchain.h

```diff
@@ -190,6 +190,45 @@ extern "C" {
 #endif
 
 /**
+ * @def OT_UNUSED_VARIABLE
+ *
+ * Suppress unused variable warning in specific toolchains.
+ *
+ */
+
+/**
+ * @def OT_UNREACHABLE_CODE
+ *
+ * Suppress Unreachable code warning in specific toolchains.
+ *
+ */
+
+#if defined(__ICCARM__)
+
+#define OT_UNUSED_VARIABLE(VARIABLE)    \
+    do                                  \
+    {                                   \
+        if (VARIABLE) {}                \
```

Comment:
@rlubos, I was consdiering to convert all `(void)var;` instances in `src/core` to use `OT_UNUSED_VARIABLE()` instead. 

Just realized that it may not work since the macro (under `ICCARM`) sort of assumes the variable can be converted to `bool` (which seems to not work with structs, references, etc.).
Is there any other way we can suppress the unused variable warning? Say, through a `_Pragma`?

How about if we do something like this? (will getting the address of the variable work for compiler).
```
do                 \
{                   \
  if (&VARIABLE == NULL) { } \
 ...
```